src/crypto/internal/boring/openssl_port_rsa.c

```diff
@@ -88,49 +88,85 @@ int _goboringcrypto_RSA_sign_raw(GO_RSA *rsa, size_t *out_len, uint8_t *out, siz
 int _goboringcrypto_RSA_sign_pss_mgf1(RSA *rsa, size_t *out_len, uint8_t *out, size_t max_out,
                       const uint8_t *in, size_t in_len, const EVP_MD *md,
                       const EVP_MD *mgf1_md, int salt_len) {
-  if (in_len != EVP_MD_size(md)) {
-    return 0;
-  }
+	EVP_PKEY_CTX *ctx;
+	EVP_PKEY *pkey;
+	size_t siglen;
+
+	pkey = EVP_PKEY_new();
+	if (!pkey)
+		return 0;
+
+	if (EVP_PKEY_set1_RSA(pkey, rsa) <= 0)
+		return 0;
+	
+	ctx = EVP_PKEY_CTX_new(pkey, NULL /* no engine */);
+	if (!ctx)
+		return 0;
+
+	int ret = 0;
+
+	if (EVP_PKEY_sign_init(ctx) <= 0)
+		goto err;
+	if (EVP_PKEY_CTX_set_rsa_padding(ctx, RSA_PKCS1_PSS_PADDING) <= 0)
+		goto err;
+	if (EVP_PKEY_CTX_set_rsa_pss_saltlen(ctx, salt_len) <= 0)
+		goto err;
+	if (EVP_PKEY_CTX_set_signature_md(ctx, md) <= 0)
+		goto err;
+	
```

Comment:
You should also call EVP_PKEY_CTX_set_rsa_mgf1_md(ctx, mgf1_md) here.

Reply:
This still applies, although current callers probably use the same for md and mgf1_md.